tiledb/sm/misc/mgc_dict.cc

```diff
@@ -37,45 +39,30 @@
 namespace tiledb {
 namespace sm {
 
-static const char magic_mgc_compressed_bytes[] = {
-#include "magic_mgc_gzipped.bin"
-};
+const tiledb::sm::ByteVecValue prepare_data() {
+  tiledb::sm::ByteVecValue expanded_buffer(magic_mgc_decompressed_size);
 
-shared_ptr<tiledb::sm::ByteVecValue> magic_dict::expanded_buffer_;
+  ConstBuffer input(magic_mgc_compressed_bytes, magic_mgc_compressed_size);
+  PreallocatedBuffer output(expanded_buffer.data(), expanded_buffer.size());
+  ZStd::ZSTD_Decompress_Context ctx;
+  ZStd::decompress(ctx, &input, &output);
 
-void* magic_dict::uncompressed_magic_dict_ = nullptr;
-
-static magic_dict magic_dict_object;
-
-magic_dict::magic_dict() {
-}
-
-void magic_dict::prepare_data() {
-  if (uncompressed_magic_dict_)
-    return;
-
-  expanded_buffer_ = make_shared<tiledb::sm::ByteVecValue>(HERE());
-  gzip_decompress(
-      expanded_buffer_,
-      reinterpret_cast<const uint8_t*>(&magic_mgc_compressed_bytes[0]));
-
-  uncompressed_magic_dict_ = expanded_buffer_.get()->data();
+  return expanded_buffer;
```

Comment:
This invokes the copy constructor. This function should likely return an rvalue of something to avoid copying.

Reply:
Mostly a theoretical question since I'm going to stop using `ByteVecValue`, but doesn't RVO apply? `ByteVecValue` doesn't have any copy constructors explicitly declared so doesn't this mean that they are implicitly declared per the rule of zero?

Reply:
Would `return std::move(expanded_buffer);` or `return std::vector<uint8_t>(std::move(expanded_buffer));` work?

Reply:
Updated to use a span.